File: LearningProject/Models/ExamDefinition.cs
namespace LearningProject.Models
{
public class ExamDefinition
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public int CategoryId { get; set; }
    public int Duration { get; set; }
    public int ExamTypeId { get; set; } 
    public ExamType? ExamType { get; set; }
    public ICollection<ExamVariant>? ExamVariants { get; set; }
    public ICollection<Exam>? Exams { get; set; }
    public Category? Category { get; set; }
}
}

File: client/src/app/core/models/ExamDefinition.ts
import { Category } from "./Category";
import { Exam } from "./Exam";
import { ExamType } from "./ExamType";
import { ExamVariant } from "./ExamVariant";

export interface ExamDefinition {
    id: number;
    examDefinitionName?: string | null;
    categoryId?: number;
    duration: number;
    examTypeId?: number;
    examType?: ExamType | null;
    examVariants?: ExamVariant[] | null;
    exams?: Exam[] | null;
    category?: Category | null;
  }
  